### DISCUSSION TITLE
Adding @click to Text objects - possible?

### DISCUSSION THREAD

**Discussion by leemthai** [Textualize/textual#4954] [General]
I can add @click to regular strings no issue:
`row_data.append(f"[@click=app.exit_app('blue')]{column.formatter.format(value)}[/]")`
But i've tried to add to Text() objects without luck. Is it possible?

Thanks

**Comment by TomJGooding** [comment]
I think the simplest way would be to use `Text.from_markup`:

```python
from rich.text import Text
from textual.app import App, ComposeResult, RenderResult
from textual.widget import Widget


class TextClick(Widget):
    def render(self) -> RenderResult:
        return Text.from_markup("[@click=hello]Click me![/]")

    def action_hello(self) -> None:
        self.notify("Hello, World!")


class ExampleApp(App):
    def compose(self) -> ComposeResult:
        yield TextClick()


if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```

**Reply by leemthai** [reply]
Perfect, thanks :)

I just noticed something quite weird. If you make the text a colour, like "Blue" you get a "Colour Preview" widget appears on top for a bit. But I can't find this in the documentation. I mean, I want to lose it. Some of my text has colours in them and I don't want the user to see colour previews  :)

Sorry, I know its unconnected with the issue.

**Reply by TomJGooding** [reply]
Sorry I'm not sure I understand what you mean by a color preview? Could you provide an example and/or a screenshot?

**Comment by leemthai** [comment]
Sure
![desktop](https://github.com/user-attachments/assets/1e067453-5767-4942-b405-bf4046360bf1)
It's not mine haha.  It appears when you hover over text in a DataTable that happens to be the name of a colour. Doesn't even have to be markup or Text, just appears when a regular string is "Blue"

And curiously it says "Colour Preview" rather than "Color Preview" so I feel like its been there since the early days.

**Reply by TomJGooding** [reply]
It looks like that's a feature of your terminal emulator, nothing to do with Textual I'm afraid!

**Reply by leemthai** [reply]
Ahhhh yes. That makes sense. And explains why I couldn't find it in the Textual documentation at all :))

Using Konsole.
